pkg/kubectl/cmd/top_node.go

```diff
@@ -32,11 +32,26 @@ import (
 
 // TopNodeOptions contains all the options for running the top-node cli command.
 type TopNodeOptions struct {
-	ResourceName string
-	Selector     string
-	NodeClient   coreclient.NodesGetter
-	Client       *metricsutil.HeapsterMetricsClient
-	Printer      *metricsutil.TopCmdPrinter
+	ResourceName    string
+	Selector        string
+	NodeClient      coreclient.NodesGetter
+	HeapsterOptions HeapsterTopOptions
+	Client          *metricsutil.HeapsterMetricsClient
+	Printer         *metricsutil.TopCmdPrinter
+}
+
+type HeapsterTopOptions struct {
+	Namespace string
+	Service   string
+	Scheme    string
+	Port      string
+}
+
+func (o *HeapsterTopOptions) Bind(flags *pflag.FlagSet) {
+	flags.StringVar(&o.Namespace, "heapster-namespace", metricsutil.DefaultHeapsterNamespace, "Namespace Heapster service is located in")
+	flags.StringVar(&o.Service, "heapster-service", metricsutil.DefaultHeapsterService, "Name of Heapster service")
+	flags.StringVar(&o.Scheme, "heapster-scheme", metricsutil.DefaultHeapsterScheme, "Scheme (http or https) to connect to Heapster as")
+	flags.StringVar(&o.Port, "heapster-port", metricsutil.DefaultHeapsterPort, "Port name in service to use")
```

Comment:
If this only supports the port name and not the number, I'd suggest to call it `--heapster-port-name` and add an example to the flag description, like "Port name in service to use, e.g. '443-tcp'".